Sources/Build/BuildPlan.swift

```diff
@@ -855,6 +855,10 @@ public final class ProductBuildDescription {
             }
         case .library(.dynamic):
             args += ["-emit-library"]
+            if buildParameters.triple.isDarwin() {
+                let relativePath = "@executable_path/\(outname.pathString)"
```

Comment:
It'd be better to use `@rpath/<dylib-name>` for dylibs so it can be found based on the application's rpath entries. In addition to that, SwiftPM can embed `@executable_path` when building executables (I thought we were already doing that) which will give you the same net effect and it'll be more flexible.

This is a good post related to this (if you haven't seen it before): https://www.mikeash.com/pyblog/friday-qa-2009-11-06-linking-and-install-names.html

Reply:
Testing it locally with `@rpath/<dylib‐name>` now. If it works, I’ll push it.

Reply:
I also had to add `-rpath @executable_path` to get it to work with `@rpath/...`. (See the fifth commit.) But then it worked.

It is pushed now. Is that what you had in mind?

Reply:
Yeah, that's what I meant by "SwiftPM can embed @executable_path when building executables"